specification/columns/contractedunitprice.md

```diff
@@ -0,0 +1,32 @@
+# Contracted Unit Price
+
+The Contracted Unit Price represents the agreed-upon unit price for a single [Pricing Unit](#pricingunit) of the associated SKU, inclusive of negotiated discounts, if present, while excluding negotiated commitment-based discounts or any other discounts. This price is denominated in the [Billing Currency](#billingcurrency). The Contracted Unit Price is commonly used for calculating savings based on negotiation activities. If negotiated discounts are not applicable, the Contracted Unit Price defaults to the [List Unit Price](#listunitprice).
+
+The ContractedUnitPrice column MUST be present in the billing data when the provider supports negotiated pricing concept. This column MUST be a Decimal within the range of non-negative decimal values, MUST conform to [Numeric Format](#numericformat) requirements, and be denominated in the BillingCurrency. When ContractedUnitPrice is present, it MUST NOT be null if [ChargeCategory](#chargecategory) is "Usage" or "Purchase" and MAY be null for other ChargeCategory values. When ContractedUnitPrice is present and not null, multiplying ContractedUnitPrice by [PricingQuantity](#pricingquantity) MUST equal [ContractedCost](#contractedcost).
+
+## Column ID
+
+ContractedUnitPrice
+
+## Display name
+
+Contracted Unit Price
+
+## Description
+
+The agreed-upon unit price for a single Pricing Unit of the associated SKU, inclusive of negotiated discounts, if present, while excluding negotiated commitment-based discounts or any other discounts.
+
+## Content Constraints
+
+| Constraint      | Value                                |
+|:----------------|:-------------------------------------|
+| Column type     | Metric                               |
+| FOCUS Essential | False                                |
+| Allows nulls    | True                                 |
+| Data type       | Decimal                              |
+| Value format    | [Numeric Format](#numericformat)     |
+| Number range    | Any valid non-negative decimal value |
```

Comment:
would we need to support a negative unit rate in the event of refunds?

Reply:
my thoughts...

**UnitPrices are always non-negative** and in case of refunds:

-  the **Quantity is negative** when ChargeCategory is Usage or Purchase and ChargeClassification is Correction (Correction indicated refund) (the` C = P * Q` math works)
-  the Quantity is null when ChargeCategory is Adjustments (the math doesn't apply)

Reply:
Discussed on 28.3.24, there is a conflict between  quantity and unit price requirements when sku price id is not present and the need to not introduce artificial variance when the 4 cost columns are not in sync... ie one column contains a value but others are NULL. we will follow up in this in #366